backend/api/Areas/Reports/Controllers/UserController.cs

```diff
@@ -0,0 +1,102 @@
+using Pims.Api.Helpers.Constants;
+using Pims.Api.Helpers.Exceptions;
+using Pims.Api.Helpers.Extensions;
+using Pims.Api.Helpers.Reporting;
+using Pims.Api.Policies;
+using EModel = Pims.Dal.Entities.Models;
+using Pims.Dal.Security;
+using Pims.Dal.Services.Admin;
+using Swashbuckle.AspNetCore.Annotations;
+using System;
+
+namespace Pims.Api.Areas.Reports.Controllers
+{
+    /// <summary>
+    /// UserController class, provides endpoints for generating reports.
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [Area("reports")]
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/[area]/users")]
+    [Route("[area]/users")]
+    public class UserController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsAdminService _pimsAdminService;
+        private readonly IMapper _mapper;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a UserController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="pimsAdminService"></param>
+        /// <param name="mapper"></param>
+        public UserController(IPimsAdminService pimsAdminService, IMapper mapper)
+        {
+            _pimsAdminService = pimsAdminService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        #region Export Users
+        /// <summary>
+        /// Exports users as CSV or Excel file.
+        /// Include 'Accept' header to request the appropriate expor -
+        ///     ["text/csv", "application/application/vnd.ms-excel", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"]
+        /// </summary>
+        /// <param name="all"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [HasPermission(Permissions.AdminUsers)]
+        [Produces(ContentTypes.CONTENT_TYPE_CSV, ContentTypes.CONTENT_TYPE_EXCELX)]
+        [ProducesResponseType(200)]
+        [SwaggerOperation(Tags = new[] { "user", "report" })]
+        public IActionResult ExportUsers(bool all = false)
+        {
+            var uri = new Uri(this.Request.GetDisplayUrl());
+            var query = Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery(uri.Query);
+            return ExportUsers(new EModel.UserFilter(query), all);
+        }
+
+        /// <summary>
+        /// Exports users as CSV or Excel file.
+        /// Include 'Accept' header to request the appropriate export -
+        ///     ["text/csv", "application/application/vnd.ms-excel", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"]
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <param name="all"></param>
+        /// <returns></returns>
+        [HttpPost("filter")]
+        [HasPermission(Permissions.AdminUsers)]
+        [Produces(ContentTypes.CONTENT_TYPE_CSV, ContentTypes.CONTENT_TYPE_EXCELX)]
+        [ProducesResponseType(200)]
```

Comment:
some of our other controllers explicitly specify known error responses as well, could add that here.